nncf/experimental/torch/fx/quantization/quantize_model.py

```diff
@@ -0,0 +1,125 @@
+# limitations under the License.
+
+from copy import deepcopy
+from typing import Optional
+
+import torch
+import torch.fx
+from torch.ao.quantization.pt2e.duplicate_dq_pass import DuplicateDQPass
+from torch.ao.quantization.pt2e.port_metadata_pass import PortNodeMetaForQDQ
+from torch.ao.quantization.pt2e.qat_utils import _fold_conv_bn_qat
+from torch.ao.quantization.pt2e.utils import _disallow_eval_train
+from torch.ao.quantization.pt2e.utils import _fuse_conv_bn_
+from torch.fx import GraphModule
+from torch.fx.passes.infra.pass_manager import PassManager
+
+import nncf
+from nncf.common.factory import NNCFGraphFactory
+from nncf.common.logging import nncf_logger
+from nncf.common.quantization.structs import QuantizationPreset
+from nncf.common.quantization.structs import QuantizationScheme
+from nncf.data import Dataset
+from nncf.experimental.torch.fx.transformations import apply_quantization_transformations
+from nncf.experimental.torch.fx.transformations import revert_quantization_transformations
+from nncf.parameters import ModelType
+from nncf.parameters import QuantizationMode
+from nncf.parameters import TargetDevice
+from nncf.quantization.advanced_parameters import AdvancedQuantizationParameters
+from nncf.quantization.advanced_parameters import QuantizationParameters
+from nncf.quantization.algorithms.post_training.algorithm import PostTrainingQuantization
+from nncf.scopes import IgnoredScope
+
+DEFAULT_RANGE_TYPE = "mean_min_max"
+
+
+def quantize_impl(
+    model: torch.fx.GraphModule,
+    calibration_dataset: Dataset,
+    mode: Optional[QuantizationMode] = None,
+    preset: Optional[QuantizationPreset] = None,
+    target_device: TargetDevice = TargetDevice.ANY,
+    subset_size: int = 300,
+    fast_bias_correction: bool = True,
+    model_type: Optional[ModelType] = None,
+    ignored_scope: Optional[IgnoredScope] = None,
+    advanced_parameters: Optional[AdvancedQuantizationParameters] = None,
+) -> torch.nn.Module:
+    """
+    Implementation of the `quantize()` method for the Torch FX backend.
+    """
+    nncf_logger.warning(
+        "Experimental Torch FX quantization backend is being used for the given torch.fx.GraphModule model."
+        " Torch FX PTQ is an experimental feature, consider using Torch or OpenVino PTQ backends"
+        " in case of errors or a poor model performance."
+    )
+    if fast_bias_correction is False:
+        raise ValueError(f"fast_bias_correction={fast_bias_correction} is not supported")
+    if target_device == TargetDevice.CPU_SPR:
+        raise nncf.InternalError("target_device == CPU_SPR is not supported")
+    if mode is not None:
+        raise ValueError(f"mode={mode} is not supported")
+
+    original_graph_meta = model.meta
+
+    copied_model = deepcopy(model)
+
+    if advanced_parameters is None:
+        advanced_parameters = AdvancedQuantizationParameters()
+    # Default quantization mode is asymmetric
```

Comment:
Why is default quantization mode asymmetric?

Reply:
Because `torch.compile` with the default backend supports only asymetric quantization scheme. Let me check `backend=openvino` works fine with the symmetric quantization and keep default configuration

Reply:
Default parameters modification is removed, openvino backend is working ok in symmetric mode